lib/extract_field/FieldExtractor.js

```diff
@@ -0,0 +1,42 @@
+const { sha256, accessNestedObjectValueByStringPath } = require("../pxutil");
+
+class FieldExtractor {
+    static SENT_THROUGH = { BODY: "body", QUERY_PARAMS: "query-param", HEADERS: "header" };
+    static CONTENT_TYPE = { JSON: "json", XML: "xml", FORM_DATA: "form-data"};
+    static CONTAINER = { "body": "body", "query-param": "query", "header": "headers"};
```

Comment:
It intend to be lowercase?

Reply:
I think I understand - it maps the SENT_TROUGH string to the name you should index to the original request to get it?
why it is not enough to keep SENT_THROUGH and for example in QUERY_PARAMS write "query" instead of "query=param" if this is the string you should use when you index

Reply:
Good question, maybe the only real necessary enum is the CONTAINER which is really a mapping of what's in the config to the name of the container field in the request object.